src/main/java/com/eternalcode/combat/fight/controller/FightUnTagController.java

```diff
@@ -30,10 +31,13 @@ void onPlayerDeath(PlayerDeathEvent event) {
         }
 
         this.announcer.sendMessage(player, this.config.messages.playerUntagged);
-        this.fightManager.untag(player.getUniqueId());
+
+        CauseOfUnTag causeOfUnTag = (killer == null ? CauseOfUnTag.PLAYER_DEATH : CauseOfUnTag.NON_PLAYER_DEATH);
+
+        this.fightManager.untag(player.getUniqueId(), causeOfUnTag);
 
         if (killer != null && this.config.settings.shouldReleaseAttacker) {
-            this.fightManager.untag(killer.getUniqueId());
+            this.fightManager.untag(killer.getUniqueId(), CauseOfUnTag.ATTACKER_RELEASE);
             this.announcer.sendMessage(killer, this.config.messages.playerUntagged);
```

Comment:
This won't work. 
I suggest to create a method:
 ```
   private CauseOfUnTag determineCauseOfUnTag(Player killer) {
        if (killer != null) {
            return (this.config.settings.shouldReleaseAttacker) ? CauseOfUnTag.ATTACKER_RELEASE : CauseOfUnTag.PLAYER_DEATH;
        }

        return CauseOfUnTag.NON_PLAYER_DEATH;
    }
```
Then the listener should look like this:
```
    @EventHandler
    void onPlayerDeath(PlayerDeathEvent event) {
        Player player = event.getEntity();
        Player killer = player.getKiller();

        if (!this.fightManager.isInCombat(player.getUniqueId())) {
            return;
        }

        this.announcer.sendMessage(player, this.config.messages.playerUntagged);

        CauseOfUnTag causeOfUnTag = this.determineCauseOfUnTag(killer);
        this.fightManager.untag(player.getUniqueId(), causeOfUnTag);
    }

    private CauseOfUnTag determineCauseOfUnTag(Player killer) {
        if (killer != null) {
            return (this.config.settings.shouldReleaseAttacker) ? CauseOfUnTag.ATTACKER_RELEASE : CauseOfUnTag.PLAYER_DEATH;
        }

        return CauseOfUnTag.NON_PLAYER_DEATH;
    }
```

Reply:
that doesn't work like that ://

We get Player Death event, if he was in FightManager he is released, just because we dont want dead people.

untag for killer is purely based on this.config.settings.shouldReleaseAttacker, that's why we do not need 3 types of CauseOfUnTag shown here:
`private CauseOfUnTag determineCauseOfUnTag(Player killer) {
        if (killer != null) {
            return (this.config.settings.shouldReleaseAttacker) ? CauseOfUnTag.ATTACKER_RELEASE : CauseOfUnTag.PLAYER_DEATH;
        }

        return CauseOfUnTag.NON_PLAYER_DEATH;
    }
    `